lumicks/pylake/nb_widgets/fd_selector.py

```diff
@@ -0,0 +1,139 @@
+            self.ranges.pop(selected[smallest_segment])
+        self.update_plot()
+
+    def handle_button_event(self, event):
+        if event.inaxes != self._axes:
+            return
+
+        # Check if we aren't using a figure widget function like zoom.
+        if event.canvas.widgetlock.locked():
+            return
+
+        event_timestamp = event.xdata * 1e9 + self.time[0]
+        if event.button == 1:
+            # Find the data point nearest to the event and determine its timestamp
+            fd_timestamp = self.time[np.argmin(np.abs(self.time - event_timestamp))]
+            if fd_timestamp:
+                self._add_point(fd_timestamp)
+        elif event.button == 3:
+            # Find whether we clicked an existing range
+            self._remove_range(event_timestamp)
+
+    def connect_click_callback(self):
+        """Connects a callback for clicking the axes"""
+        self.connection_id = self._axes.get_figure().canvas.mpl_connect('button_press_event', self.handle_button_event)
+
+    def disconnect_click_callback(self):
+        """Disconnects the clicking callback if it exists"""
+        if self.connection_id:
+            self._axes.get_figure().canvas.mpl_disconnect(self.connection_id)
+
+    def update_plot(self):
+        self._axes.clear()
+
+        for i, (t_start, t_end) in enumerate(self.ranges):
+            t_start, t_end = self.to_seconds(t_start), self.to_seconds(t_end)
+            self._axes.axvline(t_start)
+            self._axes.axvline(t_end)
+            self._axes.axvspan(t_start, t_end, alpha=0.15, color='blue')
+            plt.text((t_start + t_end) / 2, 0, i)
+
+        self.fd_curve.f.plot(linestyle='', marker='.', markersize=1)
+        self.connect_click_callback()
+
+        plt.ylabel('Force [pN]')
+        plt.xlabel('Time [s]')
+
+    @property
+    def fdcurves(self):
+        return [self.fd_curve[start:stop] for start, stop in self.ranges]
+
+
+class FdRangeSelector:
+    def __init__(self, fd_curves):
+        try:
+            import ipywidgets
+        except ImportError:
+            raise RuntimeError("This widget requires ipywidgets to be installed to work. Please install it.")
+
+        if "ipympl" not in plt.get_backend():
+            raise RuntimeError(("Please enable the widgets backend for this plot to work. You can do this by invoking "
+                                "%matplotlib widget. Please note that you may have to restart the notebook kernel for "
+                                "this to work."))
+
+        plt.figure()
+        self.axes = plt.axes()
+        self.active_plot = None
+        self.selectors = {key: FdRangeSelectorWidget(curve, self.axes, show=False) for key, curve in fd_curves.items()}
+        self.keys = [key for key, curve in fd_curves.items()]
+        ipywidgets.interact(self.update_plot,
+                            curve_idx=ipywidgets.IntSlider(min=0, max=len(fd_curves) - 1, step=1, value=0))
```

Comment:
As we've discussed, indeed, it seems that it should be simple to modify this to a Combobox with: 
```python
interact(self.update_plot, curve=self.keys);
```
and: 
```python
def update_plot(self, curve):
    if self.active_plot:
        self.active_plot.disconnect_click_callback()

    self.active_plot = self.selectors[curve]
    self.active_plot.update_plot()
```
This would make the widget easier to use.

Reply:
Definitely! Will do.